### DISCUSSION TITLE
registration token or authentication token

### DISCUSSION THREAD

**Discussion by macampo** [npalm/terraform-aws-gitlab-runner#1186] [General]
Hi,

I'm just getting started with the module so probably the next question is quite basic...

I see this in your documentation:

"The only parameter that needs to be provided is the name of a SSM parameter holding the Runner registration token and the URL of your GitLab instance". So I need to provide the runners REGISTRATION token. However, the gitlab documentation says: 

"The ability to pass a runner registration token has been [deprecated](https://docs.gitlab.com/ee/ci/runners/new_creation_workflow.html) and is planned for removal in GitLab 18.0, along with support for certain configuration arguments. This change is a breaking change. GitLab has implemented a new [GitLab Runner token architecture](https://docs.gitlab.com/ee/ci/runners/new_creation_workflow.html), which introduces a new method for registering runners and eliminates the runner registration token."

My question therefore is should I still use the Registration token option (although it is deprecated) or should I use the Authentication token (https://docs.gitlab.com/ee/security/tokens/index.html)?

**Comment by MrGiga** [comment]
The answer to this all depends on what version of Gitlab you are using and if you plan on ever updating. For future proofing I would suggest the authentication token.

**Reply by kayman-mk** [reply]
Nowadays you register the Runner manually and pass that token. This should also work for older GitLab versions.
